src/Build/Logging/BinaryLogger/BinaryLogReplayEventSource.cs

```diff
@@ -28,6 +26,11 @@ static BinaryLogReplayEventSource()
             _ = ItemGroupLoggingHelper.ItemGroupIncludeLogMessagePrefix;
         }
 
+        /// <summary>
+        /// Raised once <see cref="BuildEventArgsReader"/> is created during replaying
+        /// </summary>
+        public event Action<IBuildEventArgsReaderNotifications>? NotificationsSourceCreated;
```

Comment:
minor: often people say `event` is a legacy concept, instead people say Reactive should be used. Is it a fact? I couldn't find any articles, seems like an opinion. 

Also, we'd have to introduce Reactive libs to MSBUILD, more dependency fun. Unsure if it's worth it.

Reply:
Ah - Rx - love it :-)
Though introducing new dependencies to VS requires performance team approval, which requires strong justification (we e.g. rolled back our change leveraging faster System.IO.Hashing lib)